src/sagas/setupSaga.js

```diff
@@ -13,22 +13,33 @@ import {
   fetchComments
 } from 'sagas/threadSaga';
 import {
-  getBestRedditPost
+  getBestRedditPost,
+  pollForBestRedditPost
 } from 'effects/redditAPI';
 import {
   getUrl
 } from 'selectors/setupSelectors';
 
 export default function* onSetup() {
-  yield fork(restoreSession);
-
   yield* withThrobber(function* () {
+    yield fork(restoreSession);
+
     const url = yield select(getUrl);
     const post = yield call(getBestRedditPost, url);
 
     if (post) {
       yield put(buildAction(Actions.RedditPostIdHasChanged, post.id));
       yield* fetchComments();
+    } else {
+      yield put(buildAction(Actions.RedditPostDoNotExist));
     }
   });
 }
+
+export function* onStartPostingLinkToReddit() {
+  const url = yield select(getUrl);
+  const post = yield call(pollForBestRedditPost, url);
```

Comment:
That's exactly what you were doing, you need to find the best reddit post based on page URL and **then** load comments for this post.

The polling here happens when user wants to post the link to Reddit (when it's not there yet) and we assume that user will submit that so that's why we need poll until the post is created on Reddit so that we can allow user to post the comment.